src/main/java/org/graylog/integrations/aws/service/KinesisService.java

```diff
@@ -138,7 +136,7 @@ public HealthCheckResponse healthCheck(KinesisHealthCheckRequest request) throws
         }
 
         // Detect the type of message
-        return detectAndParseMessage(new String(payloadBytes), request.streamName(), request.logGroupName());
+        return detectAndParseMessage(new String(payloadBytes), DateTime.now().getMillis() / 1000, request.streamName(), "", "");
```

Comment:
What's the reason for using a `long` here instead of a `DateTime`? Also, why only use second precision instead of millisecond precision?

I would suggest to pass a `DateTime.now(DateTimeZone.UTC)` to `detectAndParseMessage()` and change the `timestamp()` field in `KinesisLogEntry` to `DateTime`. That way we don't have to convert the `long` back to a `DateTime` later and we don't lose any precision.

Reply:
Agreed and fixed. I also discovered that the Kinesis record has the approximate arrival time of the message in the Kinesis stream. I've changed the raw Kinesis message detection to use this instead. I figured this was better than using the time when Graylog read the message (even though it only has ms precision). Ideally, the user could provide their own timestamp with a raw message, but there's no easy solution for this.

https://github.com/Graylog2/graylog-plugin-integrations/blob/e7f0fa3fb75aea9a5b82bb72c04a7cdc26c4aaab/src/main/java/org/graylog/integrations/aws/service/KinesisService.java#L140-L143